lighthouse-core/lib/proto-preprocessor.js

```diff
@@ -23,13 +23,22 @@ function processForProto(result) {
   const reportJson = JSON.parse(result);
 
   // Clean up the configSettings
+  // Note: This is not strictly required for conversion if protobuf parsing is set to
+  //       'ignore unknown fields' in the language of conversion.
   if (reportJson.configSettings) {
-    // make sure the 'output' field is an array
-    if (reportJson.configSettings.output) {
-      if (!Array.isArray(reportJson.configSettings.output)) {
-        reportJson.configSettings.output = [reportJson.configSettings.output];
-      }
-    }
+    // Filter out unwanted fields
+    const valid = ['emulatedFormFactor', 'locale', 'onlyCategories'];
+
+    const minifiedSettings = Object.keys(reportJson.configSettings)
+      .filter(key => valid.includes(key))
+      .reduce((obj, key) => {
+        // @ts-ignore
```

Comment:
@brendankenny is there a better way to do this so that ts doesn't mind?

Reply:
trying marking `valid` as `@typeof {keyof LH.Config}`

Reply:
If I channeled my inner @paulirish, the better way would be to eliminate all non-numeric `reduce`s from the world.

I'm normally a reduce fan, but this one did confuse me for a half second and the below is actually fewer lines with no ts-ignore, for your consideration :)

```js
/** @type {Partial<LH.Config.Settings> */
const minifiedSettings = {}
for (const [key, value] of Object.entries(reportJson.configSettings) {
  if (keysToKeep.includes(key)) minifiedSettings[key] = value;
}

Reply:
Consulted @brendankenny for a solution that minimizes the ts ignores and reduces.

I :heart: reduce though.

Reply:
yeah, the issue is that currently `Object.entries` won't be well typed unless the object passed in has all the same types as its values. Since `configSettings` has a bunch of different types for its values, `[key, value]` ends up as `[string, any]`, and since `minifiedSettings` doesn't have an index, it ends up needing a `ts-ignore` for a different reason.

This solution is pretty good as an alternative, it just won't scale that well if we need more settings passed along :)

Reply:
ah my b

yeah if we don't think this list will grow. solution you ended up with is obvs best :D